gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DocumentReferenceMapper.java

```diff
@@ -0,0 +1,216 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import static org.hl7.fhir.dstu3.model.Enumerations.DocumentReferenceStatus;
+
+import static uk.nhs.adaptors.pss.translator.util.ResourceUtil.buildIdentifier;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Attachment;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.InstantType;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.v3.RCMRMT030101UK04Component;
+import org.hl7.v3.RCMRMT030101UK04Component02;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04EhrFolder;
+import org.hl7.v3.RCMRMT030101UK04NarrativeStatement;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.pss.translator.util.CompoundStatementUtil;
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+import uk.nhs.adaptors.pss.translator.util.ParticipantReferenceUtil;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class DocumentReferenceMapper {
+
+    //TODO Add custodian using the reference to Organisation that is generated for sending Practice (NIAD-2060)
+    //TODO Add file Size using the uncompressed/unencoded size of the document (NIAD-2030)
+
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-DocumentReference-1";
+    private static final String ABSENT_ATTACHMENT = "Absent Attachment";
+    private static final String PLACEHOLDER_VALUE = "GP2GP generated placeholder. Original document not available. See notes for details";
+    private static final String INVALID_CONTENT_TYPE = "Content type was not a valid MIME type";
+
+    private CodeableConceptMapper codeableConceptMapper;
+
+    public List<DocumentReference> mapToDocumentReference(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient,
+        List<Encounter> encounterList, String practiseCode) {
+        return ehrExtract.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .flatMap(ehrComposition -> ehrComposition.getComponent()
+                .stream()
+                .flatMap(this::extractAllNarrativeStatements)
+                .filter(Objects::nonNull)
+                .filter(narrativeStatement -> hasReferredToExternalDocument(narrativeStatement))
```

Comment:
Could you remove this hasReferredToExternalDocument method from both here and the ObservationCommentMapper and use the ResourceFilterUtil.isDocumentReference method? Does the exact same thing (remember to keep it negated for the ObservationCommentMapper)